spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java

```diff
@@ -302,10 +304,11 @@ public ConfigurableApplicationContext run(String... args) {
 			refreshContext(context);
 			afterRefresh(context, applicationArguments);
 			stopWatch.stop();
+			stopWatch.start("applicationReady");
```

Comment:
I think this should be moved to past the `logStarted` call

Reply:
So the `application.ready.time` is the total time from the app run()  to the end of the runners and not just the portion of time that the runners took. I will give an example to make sure we are on the same page: 

| hh:mm:ss | step |
| --- | --- |
|00:00:00|jvm is started|
|00:00:04|`run` is invoked|
|00:00:08|context has been refreshed|
|00:00:12|runners done executing|

The "base T0" time from the app perspective is `00:00:04`.  Which gives the following:

* `application.started.time`:  4s (00:00:04 -> 00:00:08)
* `application.ready.time`: 8s (00:00:04 -> 00:00:12)
* `application.ready.jvm.time`: 12s (00:00:00 -> 00:00:12)

Is that what you were thinking as well?